internal/common/foraging/deerhunt.go

```diff
@@ -0,0 +1,67 @@
+package foraging
+
+// see https://colab.research.google.com/drive/1g1tiX27Ds7FGjj4_WjFB3OLj8Fat_Ur5?usp=sharing for experiments + simulations
+
+import (
+	"math"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+type DeerHuntParams struct {
+	P   float64 // Bernoulli p variable (whether or not a deer is caught)
+	Lam float64 // Exponential lambda (scale) param for W (weight variable)
+}
+
+// DeerHunt captures the hunt participants (teams) and their resource contributions, as well as hunt params
+type DeerHunt struct {
+	Participants map[shared.ClientID]float64
+	Params       DeerHuntParams
+}
+
+// TotalInput simply sums the total group resource input of hunt participants
+func (d DeerHunt) TotalInput() float64 {
+	i := 0.0
+	for _, x := range d.Participants {
+		i += x
+	}
+	return i
+}
+
+// Hunt returns the utility from a deer hunt
+func (d DeerHunt) Hunt() float64 {
+	input := d.TotalInput()
+	decay := gameConf.ForagingConfig.IncrementalInputDecay
+	maxDeer := gameConf.ForagingConfig.MaxDeerPerHunt
```

Comment:
yeah unfortunately not a huge fan of doing `var gameConf = ...` 

Const structs are not a thing in golang (hence the reason why the config is returned via a function), and it may be accidentally changed. Code becomes immensely difficult to debug.

Reply:
So do you recommend just calling gameConf() each time instead of defining as a package var?